### PR TITLE
test/watcher

### LINKED ISSUE #691: Add tests for the custom watcher.js file
Add tests for the watcher file so it's easier to refactor in the future.

### PR DESCRIPTION
**Pull request by carlosallexandre**
## Motivation

- [x]  Add tests - _Closes #691_
  - [x] Add tests to versioned
  - [x] Add tests to assets (non markdown) versioned/root
- [x] Handle create/update/delete events emitted from `@parcel/watcher` in the `custom watcher` - _Closes #851_
- [x] Handle (events/flows/commands) within (domains/services) - _#771_
- [x] Handle appropriately the path - _Fixes #834_

### CODE REVIEW

**Comment by carlosallexandre** [comment]
There are some flaky tests that must be checked yet. :disappointed:

File: scripts/watcher.js [hunk 1]
```diff
@@ -1,69 +1,121 @@
...
+                retryEPERM(rimrafSync)(astroPath);
+                break;
+            }
+          }
+        }
+      },
+      callback
+    ),
+    {
+      ignore: [catalogDirectory],
```
Comment (boyney123) [inline scripts/watcher.js:61]: When running the code locally `npm run start:catalog` no changes are being watched. I think this is ignoring the whole directory?

**Comment by boyney123** [comment]
Thanks @carlosallexandre , will take a look. I just tried to run it all locally with `npm run start:catalog` and no changes are being watched/found. I think it's the `ignore` field that is set?

File: scripts/watcher.js (hunk 1 above)
Comment (boyney123) [reply scripts/watcher.js]: Cool I changed this, to still include the catalog directory but not to ignore files inside the project directory in the case the project DIR is inside the catalog dir, in this case for our local dev environment.

**Comment by boyney123** [comment]
@carlosallexandre when you run the tests locally do things work for you? I'm getting random errors every so often

image

**Comment by carlosallexandre** [comment]
> @carlosallexandre when you run the tests locally do things work for you? I'm getting random errors every so often

Indeed are some flaky tests. That's the reason to add the [retries](https://github.com/event-catalog/eventcatalog/blob/1cfe05f9f0ac5a35c17a2b3aab06e4071f149b1f/scripts/__tests__/watcher.spec.ts#L9). But the tests always pass here, some retry at least 1 time. 


> Thanks @carlosallexandre , will take a look. I just tried to run it all locally with `npm run start:catalog` and no changes are being watched/found. I think it's the `ignore` field that is set?

I forgot about that, i will check and return about

**Comment by boyney123** [comment]
Thanks for the effort here @carlosallexandre , I needed a lot of this to get folder structures working, so I took the watcher updates and your catalog map updates too 🙏 working great from what I can tell/test.

Last thing we need to do is merge these tests into the project, and hopefully get them less flakey somehow?

**Comment by carlosallexandre** [comment]
> Last thing we need to do is merge these tests into the project, and hopefully get them less flakey somehow?

I will work on that this week

**Comment by boyney123** [comment]
Going to close for now, we took the code from here, but just not the tests, we can add them in another PR, just doing some PR tidy up, feel free to raise again if you get round to it.

**Pull request closed without merging**